### PR TITLE
RFE: Add EasyBuild option in INSTALL.md

### PR DESCRIPTION
**Pull request by anujsharmax**
Added instructions for installing OpenCoarrays using
 - EasyBuild
 - Spack
in the Linux section of the INSTALL.md file

Resolves: #387

[links]:#
[contributing guidelines]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/CONTRIBUTING.md
[issue]: https://github.com/sourceryinstitute/OpenCoarrays/issues
[PR response img]: https://img.shields.io/issuestats/p/github/sourceryinstitute/OpenCoarrays.svg?style=flat-square
[coverage]: https://img.shields.io/codecov/c/github/sourceryinstitute/OpenCoarrays/master.svg?style=flat-square

|  Avg response time                |  coverage on master         |
|:---------------------------------:|:---------------------------:|
| ![Issue Stats][PR response img]   | ![Codecov branch][coverage] |

## Summary of changes ##

Added installation instructions for EasyBuild and Spack in the Linux section of INSTALL.md

## Rationale for changes ##

To provide alternative installation procedure for OpenCoarrays

## Additional info and certifications ##

This pull request (PR) is a:

- [ ] Bug fix
- [ ] Feature addition
- [x] Other, Please describe:

### I certify that ###

- [x] I have reviewed the [contributing guidelines] and followed the
      policies on:
      - Pull request (PR) naming to indicate work in progress (WIP),
        and to attach the PR to the appropriate bug report, or feature
        request [issue]
      - White space (no trailing white space or white space errors may
        be introduced)
      - Commenting code where it is non-obvious and non-trivial
      - Running tests locally, to ensure all of them pass
      - Maintaining or increasing test coverage
      - Logically atomic, self consistent and coherent commits
      - Commit message content
      - Waiting 24 hours before self-approving the pull request to
        give another OpenCoarrays developer a chance to review my
        proposed code

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -119,18 +119,62 @@ thread starting with [commit comment 20539810].
...
+
+# Or, To install with customisations (e.g., to install OpenCoarrays [version 1.9.0]
+# with MPICH [version default] and GCC [version 7.1.0]).
+spack install opencoarrays@1.9.0 ^mpich %gcc@7.1.0
+```
+In the previous example, it was assumed that GCC [version 7.1.0] is already installed, and is available as
+a compiler to Spack. Otherwise, [add a new compiler to Spack].
+Once installed, OpenCoarrays can be used by [loading the environment modules with Spack], e.g.
+```bash
+spack module loads --dependencies opencoarrays
```
Comment (zbeekman) [inline INSTALL.md]: is "**load_s_**" a typo here? Should it be `spack module load --dependencies opencoarrays`?

**Comment by zbeekman** [comment]
Also, at least one of the recipes is a little bit outdated; no CMake should pick up the MPI from the environment and `FC` should be set to `gfortran` and `CC` should be set to `gcc` rather than `mpifort` and `mpicc`, respectively. Also, it would be nice to indicate that GCC/GFortran >= 6 is the preferred compiler tooling in those recipes. At some point I'll try to go submit an issue, perhaps with a PR.

**Comment by zbeekman** [comment]
BTW, many thanks for this, @anujsharmax! 🙇 💯

File: INSTALL.md (hunk 1 above)
Comment (anujsharmax) [inline INSTALL.md]: It is not a typo, the command is `spack module loads`. [reference](http://spack.readthedocs.io/en/latest/module_file_support.html?highlight=module%20loads#spack-module-loads).

Also, the environment module command `module load` could be used as well.

Comment (zbeekman) [inline INSTALL.md]: In that case,

LGTM!

**Pull request merged**